compiler/x/i386/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3643,6 +3644,36 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lternaryEvaluator(TR::Node *node, T
                                    falseReg->getRegisterPair()->getHighOrder(), cg);
          }
       }
+   else if (!longCompare && condOp.isCompareForOrder() && condition->getFirstChild()->getOpCode().isIntegerOrAddress())
+      {
+      compareIntegersForOrder(condition, cg);
+      if (condOp.isCompareTrueIfEqual())
+         {
+         if (condOp.isCompareTrueIfGreater())
+            {
+            generateRegRegInstruction(CMOVL4RegReg, node, trueReg->getRegisterPair()->getLowOrder(), falseReg->getRegisterPair()->getLowOrder(), cg);
+            generateRegRegInstruction(CMOVL4RegReg, node, trueReg->getRegisterPair()->getHighOrder(), falseReg->getRegisterPair()->getHighOrder(), cg);
+            }
+         else
+            {
+            generateRegRegInstruction(CMOVG4RegReg, node, trueReg->getRegisterPair()->getLowOrder(), falseReg->getRegisterPair()->getLowOrder(), cg);
+            generateRegRegInstruction(CMOVG4RegReg, node, trueReg->getRegisterPair()->getHighOrder(), falseReg->getRegisterPair()->getHighOrder(), cg);
+            }
+         }
+      else
+         {
+         if (condOp.isCompareTrueIfGreater())
+            {
+            generateRegRegInstruction(CMOVLE4RegReg, node, trueReg->getRegisterPair()->getLowOrder(), falseReg->getRegisterPair()->getLowOrder(), cg);
+            generateRegRegInstruction(CMOVLE4RegReg, node, trueReg->getRegisterPair()->getHighOrder(), falseReg->getRegisterPair()->getHighOrder(), cg);
+            }
+         else
+            {
+            generateRegRegInstruction(CMOVGE4RegReg, node, trueReg->getRegisterPair()->getLowOrder(), falseReg->getRegisterPair()->getLowOrder(), cg);
+            generateRegRegInstruction(CMOVGE4RegReg, node, trueReg->getRegisterPair()->getHighOrder(), falseReg->getRegisterPair()->getHighOrder(), cg);
+            }
+         }      
```

Comment:
The above sequences of calls are all identical except for the opcode (across both ladders).  Can you change the if/else ladder above to just figure out the opcode, and then just do two calls to `generateRegRegInstruction` with that opcode afterwards?  This will save some DLL space.

Reply:
Addressed.